src/wallet/crypter.h

```diff
@@ -155,10 +158,19 @@ class CCryptoKeyStore : public CBasicKeyStore
             LOCK(cs_KeyStore);
             result = vMasterKey.empty();
         }
+        // fForMixing   fOnlyMixingAllowed  return
+        // ---------------------------------------
+        // true         true                result
+        // true         false               result
+        // false        true                true
+        // false        false               result
+
+        if(!fForMixing && fOnlyMixingAllowed) return true;
```

Comment:
Yep, I agree, this part is a bit confusing. I started with all possible combinations of `if`s first and then realized that they all boil down to this single one but having it alone was not very clear either, so I added this table. Probably should add few words in comments to extend it but not sure what to write here tbh. I'm open for suggestions :)